### ISSUE TITLE
some cookies not being removed when uninstalling containers

### ISSUE THREAD

**Issue by kjozwiak** [mozilla/multi-account-containers#522]
I'm not sure how many cookies this affects, but there's times where a few cookies are left behind with an associated `userContextId` even though the containers experiment has been removed. I've ran into this a few times and finally found a 100% reproducible case. In this example, there's two cookies that are left behind that belong to adnxs.com which looks like it's an online auction exchange used to sell advertising space.

This affects all versions so it's not a blocker for the 2.3.0 release.

STR (example: [https://youtu.be/Msz3tddoIC0](https://youtu.be/Msz3tddoIC0)):

* install fx and any version of containers (I used 2.3.0 but reproduced with other versions)
* once installed, open a container (in my example, I used banking)
* login into draftkings.com (should be free to create an account)
* once logged in, go back into the[ test pilot containers](https://testpilot.firefox.com/experiments/containers) page and remove containers

If you look under `about:preferences#privacy` and `cookies.sqlite`, you'll see the two following cookies which should have been removed as well:

<img width="778" alt="screen shot 2017-05-19 at 4 16 53 pm" src="https://cloud.githubusercontent.com/assets/2602313/26265311/ae121f88-3cae-11e7-87d2-923a7f4ea417.png">

I've added a video for an example which shows me logging in and removing the experiment. You'll notice when I check `about:preferences#privacy` and `cookies.sqlite`, two cookies are left behind that are still associated with `userContextId=3`.

I'm going to check to see if this is reproducible with the m-c implementation of Containers and create the appropriate bug if it's reproducible.

**Comment by jonathanKingston** [comment]
This makes me think this is a race condition in clearing the cookies here. @bakulf do you know if we clear these if the preference gets changes in m-c?

**Comment by kjozwiak** [comment]
I've managed to run into this again with a cookie from Ebay. Once I removed all the containers via the "Containers" button, I checked the cookies.sqlite database and noticed the following cookie being left behind:

<img width="753" alt="screen shot 2017-05-24 at 2 42 38 pm" src="https://cloud.githubusercontent.com/assets/2602313/26419984/5ea35702-408f-11e7-848d-30bb2dc46aef.png">

> This makes me think this is a race condition in clearing the cookies here.

I think @jonathanKingston is right... We probably have a race condition when clearing cookies. This usually happens when I start deleting several Containers at the same time. There's always a cookie or two left that will only get cleared when you clear everything from FX. Removing the Containers experiment doesn't remove the cookie either.
